lib/ruby_lsp/requests/rename.rb

```diff
@@ -0,0 +1,184 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyLsp
+  module Requests
+    # The
+    # [rename](https://microsoft.github.io/language-server-protocol/specification#textDocument_rename)
+    # request renames all instances of a symbol in a document.
+    class Rename < Request
+      extend T::Sig
+      include Support::Common
+
+      sig do
+        params(
+          global_state: GlobalState,
+          store: Store,
+          document: T.any(RubyDocument, ERBDocument),
+          params: T::Hash[Symbol, T.untyped],
+        ).void
+      end
+      def initialize(global_state, store, document, params)
+        super()
+        @global_state = global_state
+        @store = store
+        @document = document
+        @params = params
```

Comment:
nit: what do you think about storing `position` and `newName` in ivars instead? I feel it improves readability by stating what params we need directly in the constructor.

Reply:
Since `position` is the only thing we use from `@params`, should we pass it directly?

Reply:
Oh, `newName` too, so maybe not.

Reply:
No I mean still taking `params` in the constructor, but just do:

```rb
@position = params[:position]
@new_name = params[:newName]
```

Reply:
Oh, sorry I misunderstood what you said. Yeah, I agree that it's nicer like that.